superset/db_engine_specs/gsheets.py

```diff
@@ -136,6 +153,89 @@ def extra_table_metadata(
 
         return {"metadata": metadata["extra"]}
 
+    @staticmethod
+    def is_oauth2_enabled() -> bool:
+        """
+        Return if OAuth2 is enabled for GSheets.
+        """
+        return "Google Sheets" in current_app.config["DATABASE_OAUTH2_CREDENTIALS"]
+
+    @staticmethod
+    def get_oauth2_authorization_uri(state: OAuth2State) -> str:
+        """
+        Return URI for initial OAuth2 request.
+
+        https://developers.google.com/identity/protocols/oauth2/web-server#creatingclient
+        """
+        config = current_app.config["DATABASE_OAUTH2_CREDENTIALS"]["Google Sheets"]
+        redirect_uri = current_app.config.get(
+            "DATABASE_OAUTH2_REDIRECT_URI",
+            state["default_redirect_uri"],
+        )
+
+        encoded_state = jwt.encode(
+            payload=state,
+            key=current_app.config["SECRET_KEY"],
+            algorithm=current_app.config["DATABASE_OAUTH2_JWT_ALGORITHM"],
```

Comment:
should we put an expiration on this jwt?

Reply:
Yeah, good idea!